File: var/www/cgi-bin/form-delete.rb
#!/usr/bin/ruby

require "cgi"

cgi = CGI.new

was_file_deleted = false

if File.exist?(cgi["filename"]) && !File.directory?(cgi["filename"]) && cgi["filename"].include?("/var/www/files/")
	File.delete(cgi["filename"])

	was_file_deleted = !File.exist?(cgi["filename"])
end

puts 'HTTP/1.0 200 OK'
puts 'Content-type: text/html'
puts ''

puts '<!DOCTYPE html>'
puts '<html lang="en">'
puts '	<head>'
puts '		<title>42 Webserver - Form Submission</title>'
puts '		<link rel="stylesheet" href="/index.css">'
puts '		<script src="/index.js"></script>'
puts '	</head>'
puts '	<body>'
puts '		<div class="page-wrapper">'
puts '			<div class="container">'
puts '				<section>'
puts '					<h1 class="title">'
puts "						File #{was_file_deleted ? 'deleted' : 'not deleted'}"
puts '					</h1>'
puts '				</section>'
puts '			</div>'
puts '		</div>'
puts '	</body>'
puts '</html>'

File: var/www/cgi-bin/form-delete.py
#!/usr/bin/python

import cgi
import os

try:
	form = cgi.FieldStorage()

	was_file_deleted = False

	if "filename" in form and form["filename"] and os.path.exists(form["filename"]):
		try:
			os.remove(form["filename"])
			was_file_deleted = True
		except Exception as e:
			was_file_deleted = False

	print('HTTP/1.0 200 OK')
	print('Content-type: text/html')
	print('')

	print('<!DOCTYPE html>')
	print('<html lang="en">')
	print('	<head>')
	print('		<title>42 Webserver - Form Submission</title>')
	print('		<link rel="stylesheet" href="/index.css">')
	print('		<script src="/index.js"></script>')
	print('	</head>')
	print('	<body>')
	print('		<div class="page-wrapper">')
	print('			<div class="container">')
	print('				<section>')
	print('					<h1 class="title">')
	print("						File {}".format('deleted' if was_file_deleted else 'not deleted'))
	print('					</h1>')
	print('				</section>')
	print('			</div>')
	print('		</div>')
	print('	</body>')
	print('</html>')
except Exception as e:
	print("HTTP/1.0 500 Internal Server Error")
